spec.md

```diff
@@ -936,6 +937,20 @@ module Top:
   node consumer_debug = read(c.out.cref); ; Consumer-side signal
 ```
 
+## Property Types
+
+FIRRTL property types represent information about the circuit that is not
+hardware. This is useful to capture domain specific knowledge and design intent
+alongside the hardware description within the same FIRRTL.
+
+Property types cannot affect hardware. They only exist to augment the hardware
+description with extra information.
+
+Handling of property types is completely implementation defined. A valid FIRRTL
+compiler implementation may do anything with property types as long as the
+existence of property types does not affect the emitted hardware description.
+For example, it is valid to drop property types from the IR completely.
+
```

Comment:
This is good, though is a bit cagey. Are there any examples that we can include in this PR to make things more obvious. E.g., `BigInt` and `String` type?

Or: are you planning to split this out in a very fine-grained way over multiple PRs?

Reply:
I was planning to multiple fine grained PRs, so the next one would be to define a concrete property type like BigInt. If you think it makes more sense to have examples in the first PR, I can do that. Or, if it would help, I can get my act together and send the next PR based on this one.

Reply:
Multiple PRs are fine. 👍

Reply:
in terms of impacting hardware, and in absence of any description of dedup, is it legal to use properties to decide whether to dedup modules or not... do we need to say anything? Or maybe we could lightly weaken this statement by saying "hardware functionality" vs "hardware"

Reply:
I am now using the phrasing "cannot affect hardware functionality or the hardware ABI". Still somewhat vague, since we don't define "hardware functionality", but let me know what you all think.